src/integrations/rapier/fluids_pipeline.rs

```diff
@@ -152,108 +153,110 @@ impl<'a> CouplingManager for ColliderCouplingManager<'a> {
         fluids: &mut [Fluid],
         boundaries: &mut BoundarySet,
     ) {
+        let bodies = &self.bodies;
         for (collider, coupling) in &mut self.coupling.entries {
             if let (Some(collider), Some(boundary)) = (
                 self.colliders.get(*collider),
                 boundaries.get_mut(coupling.boundary),
             ) {
                 // Update the boundary's ability to receive forces.
-                let body = self.bodies.get(collider.parent());
-                if let Some(body) = body {
-                    if !body.is_dynamic() {
-                        boundary.forces = None;
-                    } else {
-                        boundary.forces = Some(RwLock::new(Vec::new()));
-                        boundary.clear_forces(true);
+                if let Some(body) = collider.parent().map(|p| bodies.get(p)) {
+                    if let Some(body) = body {
```

Comment:
Use `.and_then` to chain the `Options` into a single option:
```suggestion
                if let Some(body) = collider.parent().and_then(|p| bodies.get(p)) {
```